code/components/extra-natives-rdr3/src/ShapeTestNatives.cpp

```diff
@@ -0,0 +1,98 @@
+#include <StdInc.h>
+#include <ScriptEngine.h>
+
+#include <Hooking.h>
+#include <scrEngine.h>
+#include <EntitySystem.h>
+#include <ScriptSerialization.h>
+
+struct ShapeTestResult
+{
+	uint32_t shapeTestHandle;
+	bool hit;
+	scrVector endCoords;
+	scrVector surfaceNormal;
+	uint64_t entityHit;
+};
+
+class ResultData
+{
+public:
+	char pad_0000[108]; // 0x0000
+	uint8_t materialIndex; // 0x006C
+}; // Size: 0x006D
+
+class CScriptShapeTestResult
+{
+public:
+	char pad_0000[36]; // 0x0000
+	uint32_t state; // 0x0024
+	char pad_0028[8]; // 0x0028
+	class ResultData* data; // 0x0030
+}; // Size: 0x0038
+
+template<std::size_t Index, typename ReturnType, typename... Args>
+inline ReturnType call_virtual(void* instance, Args... args)
+{
+	using Fn = ReturnType(__thiscall*)(void*, Args...);
+
+	auto function = (*reinterpret_cast<Fn**>(instance))[Index];
+	return function(instance, args...);
+}
+
+static hook::cdecl_stub<CScriptShapeTestResult*(uint64_t)> g_getCScriptShapeTestResult([]()
+{
+	return hook::get_call(hook::get_pattern("E9 ? ? ? ? 90 F3 0F 8B 05 1A 2D BB"));
```

Comment:
This pattern is invalid on 1311 and 1355. You can not rely on layout outside of the function you're hooking data from, there's 99.9% of chance it will be different in a next build. My version:
```cpp
return hook::get_call(hook::get_pattern("E8 ? ? ? ? 4C 8B C0 48 85 C0 75 04"));
```